// ng-input-password/ng-input-password.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var forms_1 = require("@angular/forms");
var noop = function () {
};
var CUSTOM_INPUT_CONTROL_VALUE_ACCESSOR = {
    provide: forms_1.NG_VALUE_ACCESSOR,
    useExisting: core_1.forwardRef(function () { return NgInputPasswordComponent; }),
    multi: true
};
/**
 * Generated class for the NgInputPasswordComponent component.
 *
 * See https://angular.io/docs/ts/latest/api/core/index/ComponentMetadata-class.html
 * for more info on Angular Components.
 */
var NgInputPasswordComponent = (function () {
    function NgInputPasswordComponent() {
        this.placeholder = 'Password';
        this.color = 'primary';
        this.hideColor = 'dark';
        this.icon = 'eye';
        this.label = '';
        this.labelStyle = '';
        this.isShow = false;
        this.typeInput = 'password';
        //The internal data model
        this.innerValue = '';
        //Placeholders for the callbacks which are later providesd
        //by the Control Value Accessor
        this.onTouchedCallback = noop;
        this.onChangeCallback = noop;
    }
    NgInputPasswordComponent_1 = NgInputPasswordComponent;
    NgInputPasswordComponent.prototype.hasLabel = function () {
        return this.label != '';
    };
    NgInputPasswordComponent.prototype.isLabelStyle = function (style) {
        return this.labelStyle == style;
    };
    /**
     *
     */
    NgInputPasswordComponent.prototype.showHidePass = function () {
        this.isShow = !this.isShow;
        if (this.typeInput == 'password') {
            this.typeInput = 'text';
        }
        else {
            this.typeInput = 'password';
        }
    }; // showHidePass
    /**
     *
     */
    NgInputPasswordComponent.prototype.toggleColor = function () {
        if (this.isShow)
            return this.color;
        else
            return this.hideColor;
    }; // toggleColor
    Object.defineProperty(NgInputPasswordComponent.prototype, "value", {
        //get accessor
        get: function () {
            return this.innerValue;
        },
        //set accessor including call the onchange callback
        set: function (v) {
            if (v !== this.innerValue) {
                this.innerValue = v;
                this.onChangeCallback(v);
            }
        },
        enumerable: true,
        configurable: true
    });
    ;
    //Set touched on blur
    NgInputPasswordComponent.prototype.onBlur = function () {
        this.onTouchedCallback();
    };
    //From ControlValueAccessor interface
    NgInputPasswordComponent.prototype.writeValue = function (value) {
        if (value !== this.innerValue) {
            this.innerValue = value;
        }
    };
    //From ControlValueAccessor interface
    NgInputPasswordComponent.prototype.registerOnChange = function (fn) {
        this.onChangeCallback = fn;
    };
    //From ControlValueAccessor interface
    NgInputPasswordComponent.prototype.registerOnTouched = function (fn) {
        this.onTouchedCallback = fn;
    };
    __decorate([
        core_1.Input('placeholder'),
        __metadata("design:type", Object)
    ], NgInputPasswordComponent.prototype, "placeholder", void 0);
    __decorate([
        core_1.Input('color'),
        __metadata("design:type", String)
    ], NgInputPasswordComponent.prototype, "color", void 0);
    __decorate([
        core_1.Input('hideColor'),
        __metadata("design:type", String)
    ], NgInputPasswordComponent.prototype, "hideColor", void 0);
    __decorate([
        core_1.Input('icon'),
        __metadata("design:type", String)
    ], NgInputPasswordComponent.prototype, "icon", void 0);
    __decorate([
        core_1.Input('label'),
        __metadata("design:type", String)
    ], NgInputPasswordComponent.prototype, "label", void 0);
    __decorate([
        core_1.Input('labelStyle'),
        __metadata("design:type", String)
    ], NgInputPasswordComponent.prototype, "labelStyle", void 0);
    NgInputPasswordComponent = NgInputPasswordComponent_1 = __decorate([
        core_1.Component({
            selector: 'ng-input-password',
            template: "\n    <ion-item>\n    <ion-label *ngIf=\"hasLabel() && labelStyle == 'fixed' \" fixed >{{label}}</ion-label>\n    <ion-label *ngIf=\"hasLabel() && labelStyle == 'floating' \" floating >{{label}}</ion-label>\n    <ion-input [type]=\"typeInput\" [(ngModel)]=\"value\" [placeholder]=\"placeholder\"  ></ion-input>\n    <button ion-button clear [color]=\"toggleColor()\" type=\"button\" item-right (click)=\"showHidePass()\"> <ion-icon [name]=\"icon\"> </ion-icon> </button>\n    </ion-item>\n    ",
            providers: [
                {
                    provide: forms_1.NG_VALUE_ACCESSOR,
                    useExisting: core_1.forwardRef(function () { return NgInputPasswordComponent_1; }),
                    multi: true
                }
            ]
        }),
        __metadata("design:paramtypes", [])
    ], NgInputPasswordComponent);
    return NgInputPasswordComponent;
    var NgInputPasswordComponent_1;
}());
exports.NgInputPasswordComponent = NgInputPasswordComponent;
